pkg/detectors/tableau/tableau.go

```diff
@@ -0,0 +1,350 @@
+	// Build the verification URL
+	verifyURL := fmt.Sprintf("https://%s/api/3.26/auth/signin", endpoint)
+
+	// Prepare metadata early - before any potential errors
+	extraData := map[string]string{
+		"verification_endpoint": verifyURL,
+		"verification_method":   "tableau_pat_auth",
+		"tableau_endpoint":      endpoint,
+	}
+
+	// Rest of your verification logic...
+	authReq := TableauAuthRequest{
+		Credentials: TableauCredentials{
+			PersonalAccessTokenName:   tokenName,
+			PersonalAccessTokenSecret: tokenSecret,
+			Site:                      map[string]interface{}{},
+		},
+	}
+
+	// Marshal to JSON
+	jsonData, err := json.Marshal(authReq)
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to marshal auth request: %v", err)
+	}
+
+	// Create HTTP request
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, verifyURL, bytes.NewBuffer(jsonData))
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to create request: %v", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+
+	// Execute request
+	resp, err := client.Do(req)
+	if err != nil {
+		// Check if it's a DNS/network error
+		if strings.Contains(err.Error(), "no such host") ||
+			strings.Contains(err.Error(), "dial tcp") ||
+			strings.Contains(err.Error(), "connection refused") {
+			extraData["verification_status"] = "invalid_endpoint"
+			extraData["network_error"] = "true"
+			return false, extraData, nil // No error, just invalid endpoint
+		}
+		return false, nil, fmt.Errorf("request failed: %v", err)
+	}
+	defer func() {
+		_, _ = io.Copy(io.Discard, resp.Body)
+		_ = resp.Body.Close()
+	}()
+
+	// Read the response
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to read response body: %v", err)
+	}
+
+	// Add HTTP status to metadata
+	extraData["http_status"] = fmt.Sprintf("%d", resp.StatusCode)
+
+	// Status code handling...
+	switch resp.StatusCode {
+	case http.StatusOK:
+		var authResp TableauAuthResponse
+		if err := json.Unmarshal(bodyBytes, &authResp); err == nil {
+			if token := authResp.Credentials.Token; token != "" {
+				extraData["auth_token_received"] = "true"
+				extraData["site_id"] = authResp.Credentials.Site.ID
+				extraData["user_id"] = authResp.Credentials.User.ID
+				extraData["verification_status"] = "valid"
+				return true, extraData, nil
+			}
+		}
+		// Fallback success on 200
+		extraData["verification_status"] = "valid"
+		return true, extraData, nil
+
+	case http.StatusUnauthorized:
+		extraData["verification_status"] = "invalid"
+		return false, extraData, nil
+
+	case http.StatusBadRequest:
+		extraData["verification_status"] = "invalid_format"
+		return false, extraData, nil
+
+	case http.StatusForbidden:
+		extraData["verification_status"] = "insufficient_permissions"
+		return false, extraData, nil
+
```

Comment:
All of these statuses can be handled together in one case and it's not necessary to add verification status in the extraData